include/wx/hwinfo.h

```diff
@@ -0,0 +1,40 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/hwinfo.h
+// Purpose:     declaration of the wxSystemHardwareInfo class
+// Author:      Blake Madden
+// Created:     2024-07-09
+// Copyright:   (c) 2024 Blake Madden
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_SYSTEMHWINFO_H_
+#define _WX_SYSTEMHWINFO_H_
+
+#include "wx/string.h"
+#include "wx/utils.h"
+#include "wx/power.h"
+
+// ----------------------------------------------------------------------------
+// wxSystemHardwareInfo
+// ----------------------------------------------------------------------------
+
+// Information about the system's hardware
+class WXDLLIMPEXP_BASE wxSystemHardwareInfo
+{
+public:
+
+    static int GetCPUCount();
+
+    static wxString GetCPUArchitectureName();
+    static wxString GetNativeCPUArchitectureName();
+    static bool IsRunningNatively();
+
+    static wxMemorySize GetPhysicalMemory();
+    static wxMemorySize GetFreePhysicalMemory();
```

Comment:
Sorry, I don't remember if we discussed this already, but do we really need "Physical" in the names of these functions? I think just `Get{Total,Free}Memory()` would be good enough.

Also, I wonder if we need `GetUsedMemory()`?